In [5]:
import pandas as pd
import numpy as np

In [31]:
#Q1
URL = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(URL)
df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [29]:
#Q2
# Mengakses indeks dari tuple df.shape
jumlah_baris = df.shape[0]
jumlah_kolom = df.shape[1]

# Menampilkannya dengan f-string agar mudah dibaca
print(f"Total Baris : {jumlah_baris}")
print(f"Total Kolom : {jumlah_kolom}")

Total Baris : 891
Total Kolom : 12


Hasil : Data tersebut terdiri dari 819 Baris, dan 12 Kolom

In [23]:
#Q3
total_kosong = df.isnull().sum()

# Memfilter HANYA kolom yang memiliki nilai kosong (lebih dari 0)
kolom_kosong = total_kosong[total_kosong > 0]

# Menghitung persentase dari total baris (len(df))
persentase_kosong = (kolom_kosong / len(df)) * 100

# Menggabungkan hasil ke dalam tabel (DataFrame) agar rapi
df_missing_info = pd.DataFrame({
    'Total Kosong': kolom_kosong,
    'Persentase (%)': persentase_kosong.round(2) # round(2) untuk membulatkan 2 desimal
})

# Mengurutkan dari persentase yang paling besar ke kecil
df_missing_info = df_missing_info.sort_values(by='Persentase (%)', ascending=False)

# Menampilkan hasil
print(f"Ditemukan {len(df_missing_info)} kolom yang memiliki nilai kosong:\n")
print(df_missing_info)

Ditemukan 3 kolom yang memiliki nilai kosong:

          Total Kosong  Persentase (%)
Cabin              687           77.10
Age                177           19.87
Embarked             2            0.22


Hasil : Data yang Kosong yaitu Cabin(77.10%), Age(19.87), Embarked (0.22).

In [32]:
#Q4
persentase_selamat = df['Survived'].mean() * 100

print(f"Persentase Penumpang Selamat: {persentase_selamat:.2f}%")

Persentase Penumpang Selamat: 38.38%


Hasil : Persentase Penumpang Selamat: 38.38%

In [33]:
#Q5
# Memfilter wanita di Pclass 1
wk1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]

# Print harus di baris baru
print("Persentase Selamat (Wanita Kelas 1):", wk1['Survived'].mean() * 100, "%")
print("Rata-rata Umur (Wanita Kelas 1):", wk1['Age'].mean(), "tahun")

Persentase Selamat (Wanita Kelas 1): 96.80851063829788 %
Rata-rata Umur (Wanita Kelas 1): 34.61176470588235 tahun


Hasil : Rata-rata usia Penumpang Kelas 1 ( 34.61 Tahun )

In [28]:
#Q6
# Mengurutkan DataFrame 'hasil' berdasarkan kolom 'selamat' dari yang paling besar (descending)
hasil_diurutkan = hasil.sort_values(by='selamat', ascending=False)

print("Tingkat Keselamatan Penumpang Berdasarkan Kelas:")
print(hasil_diurutkan)

Tingkat Keselamatan Penumpang Berdasarkan Kelas:
        total  selamat
Pclass                
1         216    0.630
2         184    0.473
3         491    0.242


Hasil : Penumpang dengan Keselamatan tertinggi

In [35]:
df_processed['Age'] = df_processed['Age'].fillna(df_processed['Age'].median())
df_processed['AgeCategory'] = np.where(df_processed['Age'] < 18, 'Anak-anak', 'Dewasa')

# Kelompokkan berdasarkan Pclass, AgeCategory, dan Survived
summary_data = df_processed.groupby(['Pclass', 'AgeCategory', 'Survived']).size().unstack(fill_value=0)

# Pastikan nama kolom berurutan (0 = Tidak Selamat, 1 = Selamat)
summary_data.columns = ['Tidak Selamat', 'Selamat']
summary_data['Total'] = summary_data['Tidak Selamat'] + summary_data['Selamat']

# Hitung persentase keselamatan
summary_data['% Selamat'] = (summary_data['Selamat'] / summary_data['Total'] * 100).round(2)
summary_data['% Tidak Selamat'] = (summary_data['Tidak Selamat'] / summary_data['Total'] * 100).round(2)

print("Analisis Keselamatan Berdasarkan Kelas dan Kategori Usia:")
# Gunakan print biasa jika tidak jalan di environment non-Jupyter.
# Jika di Jupyter/Colab, display() sudah tepat.
display(summary_data)

print("\nRata-rata Tingkat Keselamatan (Mean Survival Rate):")
display(df_processed.groupby(['Pclass', 'AgeCategory'])['Survived'].mean().unstack().round(3))

Analisis Keselamatan Berdasarkan Kelas dan Kategori Usia:


Tidak Selamat  Selamat  Total  % Selamat  % Tidak Selamat
Pclass AgeCategory                                                           
1      Anak-anak                1       11     12      91.67             8.33
       Dewasa                  79      125    204      61.27            38.73
2      Anak-anak                2       21     23      91.30             8.70
       Dewasa                  95       66    161      40.99            59.01
3      Anak-anak               49       29     78      37.18            62.82
       Dewasa                 323       90    413      21.79            78.21


Rata-rata Tingkat Keselamatan (Mean Survival Rate):


AgeCategory,Anak-anak,Dewasa
Pclass,,
1,0.917,0.613
2,0.913,0.410
3,0.372,0.218


Saya Mencoba untuk Menampilkan Data Analisis Keselamatan berdasarkan kelas dan kategori usia, selain itu saya lampirkan rata-rata tingkat keselamatan.